src/Sarif.Multitool/Rules/SARIF1008.PhysicalLocationPropertiesMustBeConsistent.cs

```diff
@@ -18,15 +21,130 @@ public class PhysicalLocationPropertiesMustBeConsistent : SarifValidationSkimmer
 
         protected override IEnumerable<string> MessageResourceNames => new string[]
         {
-            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionRequiresRegion_Text)
+            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionRequiresRegion_Text),
+            nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionMustBeProperSupersetOfRegion_Text)
         };
 
         protected override void Analyze(PhysicalLocation physicalLocation, string physicalLocationPointer)
         {
-            if (physicalLocation.ContextRegion != null && physicalLocation.Region == null)
+            if (physicalLocation.ContextRegion == null)
+            {
+                return;
+            }
+
+            if (physicalLocation.Region == null)
             {
                 LogResult(physicalLocationPointer, nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionRequiresRegion_Text));
+                return;
+            }
+
+            if (!IsRegionProperSuperset(physicalLocation.ContextRegion, physicalLocation.Region))
+            {
+                LogResult(physicalLocationPointer, nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionMustBeProperSupersetOfRegion_Text));
+            }
+        }
+
+        private static bool IsRegionProperSuperset(Region superRegion, Region subRegion)
+        {
+            if (IsLineColumnBasedTextRegion(superRegion) &&
+                IsLineColumnBasedTextRegion(subRegion) &&
+                !IsLineColumnBasedTextRegionProperSuperset(superRegion, subRegion))
+            {
+                return false;
+            }
+
+            if (IsOffsetBasedTextRegion(superRegion) &&
+                IsOffsetBasedTextRegion(subRegion) &&
+                !IsOffsetBasedTextRegionProperSupetSet(superRegion, subRegion))
+            {
+
+                return false;
             }
+
+            if (IsBinaryRegion(superRegion) &&
+                IsBinaryRegion(subRegion) &&
+                !IsBinaryRegionProperSuperset(superRegion, subRegion))
+            {
+                return false;
+            }
+
+            // if we reach here, the region and context region have been expressed as different property sets,
+            // and it is not possible to judge validity without looking at the actual content.
+            // It is a potential false negative.
+            return true;
+
+        }
+
+        private static bool IsBinaryRegionProperSuperset(Region superRegion, Region subRegion)
+        {
+            if (superRegion.ByteOffset > subRegion.ByteOffset)
+            {
+                return false;
+            }
+
+            if (superRegion.ByteOffset == subRegion.ByteOffset && superRegion.ByteLength <= subRegion.ByteLength)
+            {
+                return false;
```

Comment:
>false [](start = 23, length = 5)

This misses the case where superRegion starts _before_ subRegion and ends after. You need to compute endOffset = startOffset + length for each region, then compare the starts and ends.

Reply:
gosh u r right.

---
In reply to: [443872755](https://github.com/microsoft/sarif-sdk/pull/1925#discussion_r443872755) [](ancestors = 443872755)